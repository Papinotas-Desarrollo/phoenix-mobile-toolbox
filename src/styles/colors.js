const colors = {
  menuGrey: '#7D7D7D',
  bottomMenu: 'rgba(249, 249, 249, 0.9)',
  lightGrey: '#F7F7F7',
  black03: '#00000056',
  iconBlack: '#707070',
  papinotasBlue: '#4A90E2',
  backgroundLightBlue: '#4A90E220',
  papinotasOrange: '#FF9500',
  menuGray: '#8E8E93',
  lineGray: '#D1D1D6',
  buttons: {
    inactive: '#8E8E93',
    active: '#007AFF',
  },
  white: '#FFFFFF',
  black: '#000000',
  green: '#1C6900',
  red: '#FF0000',
  transparent: 'transparent',
  orange: '#f5a623',
  mediumGray: '#9a9a9a',
  blueFilter: '#7dafef',
  backgroundGrey: '#3690E9',
};

export default colors;
