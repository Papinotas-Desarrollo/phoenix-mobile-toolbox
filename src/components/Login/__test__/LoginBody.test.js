import * as React from 'react';
import { TouchableHighlight } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { LoginBody } from '../';
import colors from '../../../styles/colors';

describe('Snapshot test', () => {
  it('With props snapshot', () => {
    const props = {
      bodyTitle: 'bodyTitle',
      bodySubtitle: 'bodySubtitle',
      beforeInputText: 'beforeInputText',
      firstInputPlaceholder: 'firstInputPlaceholder',
      secondInputPlaceholder: 'secondInputPlaceholder',
      onChangeEmail: jest.fn(),
      onChangePassword: jest.fn(),
    };
    const tree = renderer.create(<LoginBody {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Proptype check', () => {
  it('Callback proptypes defined', () => {
    expect(LoginBody.propTypes.onChangeEmail).toBeDefined();
    expect(LoginBody.propTypes.onChangePassword).toBeDefined();
  });
});
describe('Functionality', () => {
  it('Show password', () => {
    const props = {
      bodyTitle: '',
      bodySubtitle: '',
      beforeInputText: '',
      firstInputPlaceholder: '',
      secondInputPlaceholder: '',
      onChangeEmail: jest.fn(),
      onChangePassword: jest.fn(),
    };
    const wrapper = shallow(<LoginBody {...props} />)
    expect(wrapper.find(Icon).at(0).props().color).toBe(colors.papinotasBlue);
    wrapper.find(TouchableHighlight).at(0).props().onPress();
    expect(wrapper.find(Icon).at(0).props().color).toBe(colors.papinotasOrange);
  });
});
