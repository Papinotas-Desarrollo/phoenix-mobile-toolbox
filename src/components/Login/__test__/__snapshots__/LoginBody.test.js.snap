// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test With props snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#4A90E2",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "600",
          "marginBottom": 13,
          "marginLeft": 13,
        }
      }
    >
      bodyTitle
    </Text>
  </View>,
  <View>
    <Text
      style={
        Object {
          "color": "#FF9500",
          "fontSize": 20,
          "marginBottom": 5,
          "marginLeft": 13,
          "marginTop": 9,
        }
      }
    >
      bodySubtitle
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 13,
          "marginHorizontal": 13,
        }
      }
    >
      beforeInputText
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize={false}
      keyboardType="email-address"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="firstInputPlaceholder"
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 10,
          "marginLeft": 13,
          "marginRight": 16,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#D1D1D6",
            "borderBottomWidth": 2,
            "marginBottom": 42,
            "marginHorizontal": 13,
          },
          null,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 13,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="secondInputPlaceholder"
        secureTextEntry={true}
        style={
          Object {
            "flex": 0.8,
            "fontSize": 16,
            "marginBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 0.2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#4A90E2",
                "fontSize": 30,
              },
              null,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Óêó
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#D1D1D6",
            "borderBottomWidth": 2,
            "marginBottom": 42,
            "marginHorizontal": 13,
          },
          null,
        ]
      }
    />
  </View>,
]
`;
